PROJECT_CHARTER.md

```diff
@@ -0,0 +1,51 @@
+
+## Section 1: Scope
+
+`jest` is primarily a test framework and test runner, focused on making it easy to test any JS code, either written for the server, the client or both. The Jest project consists of many different smaller modules, such as the assertion library, test runner, diffing algorithm, object serializer, task manager and more. The project is designed to be modular, so that it can be used in a variety of different environments and for different purposes.
+
+## Section 2: Relationship with OpenJS Foundation CPC.
+
+Technical leadership for the projects within the [OpenJS Foundation][openjs foundation] is delegated to the projects through their project charters by the [OpenJS Foundation Cross-Project Council](https://openjsf.org/about/governance/) (CPC). In the case of the `jest` project, it is delegated to the [`jest` Maintainers](GOVERNANCE.md#maintainers) (the “Maintainers”). The OpenJS Foundation's business leadership is the Board of Directors (the “Board”).
+
+This `jest` Charter reflects a carefully constructed balanced role for the Maintainers and the CPC in the governance of the OpenJS Foundation. The charter amendment process is for the Maintainers to propose changes using simple majority of the full Maintainers, the proposed changes being subject to review and approval by the CPC. The CPC may additionally make amendments to the project charter at any time, though the CPC will not interfere with day-to-day discussions, votes or meetings of the Maintainers.
+
+### 2.1 Other Formal Project Relationships
+
+Section Intentionally Left Blank
+
+## Section 3: `jest`'s Maintainers Governing Body
+
+`jest` is governed by its [maintainers](GOVERNANCE.md#maintainers).
+
+## Section 4: Roles & Responsibilities
+
+The roles and responsibilities of `jest`'s Maintainers are described in [GOVERNANCE.md](./GOVERNANCE.md).
+
+### Section 4.1 Project Operations & Management
+
+Section Intentionally Left Blank
+
+### Section 4.2: Decision-making, Voting, and/or Elections
+
+Section Intentionally Left Blank
+
+### Section 4.3: Other Project Roles
+
+Section Intentionally Left Blank
+
+## Section 5: Definitions
+
+- _Outside Contributors_: contribute code or other artifacts, but do not have the right to commit to the codebase. Contributors work with the project’s maintainers to have code committed to the code base. An Outside Contributor may be promoted to a Contributor or Maintainer by the Maintainers. Outside Contributors should rarely be encumbered by the Maintainers and never by the CPC or OpenJS Foundation Board.
+
+- _Contributors_: Contributors have triaging rights and commit access to the repository.
+
+- _Maintainers_: Contributors with any kind of decision-making authority in the project, and access to publishing and CI setup.
+
+[openjs foundation]: https://openjsf.org
```

Comment:
I'd lighten this section a bit and avoid anything that feels like an implementation detail of the charter. For example, stating that contributors have commit access to the repository forces you to go through a CPC charter modification process the day you want change that.

Reply:
specifically the commit access, or triaging etc as well? I essentially wanted to capture the different accesses GH provides

<img width="312" alt="image" src="https://github.com/jestjs/jest/assets/1404810/67e3f8c4-2689-430a-8245-bec85c8ad1dd">

But with merging "maintainer" and "admin". I guess just having triaging makes sense. And actually what I want in the future, as I wanna set up CI publishing, and that would mean push access is enough to trigger a release.

Reply:
I'd frame this outside of GH roles and would avoid the internal/external terminology for contributors (which tends to be tied to differentiating employee and non-employee contributors in corporate open source settings). You also don't really need to define non-core contributors. So you could go with something very light-weight, for example:

Maintainers: responsible for the technical direction of the project and for release management.
Core contributors: Maintainers delegate project responsibilities to core contributors, as documented in governance.md.

Reply:
@SimenB does adjusting this section to just be the following make sense?

Maintainers: responsible for the technical direction of the project and for release management.
Core contributors: Maintainers delegate project responsibilities to core contributors, as documented in governance.md.

Reply:
[`302b719` (#14948)](https://github.com/jestjs/jest/pull/14948/commits/302b719c79781fcbd3a321af38dfae12b380d6e3)